src/test/java/com/mysql/cj/util/RdsUtilsTests.java

```diff
@@ -0,0 +1,200 @@
+    assertEquals(chinaExpectedHostPattern, target.getRdsInstanceHostPattern(chinaRegionCustomDomain));
+
+    final String oldChinaExpectedHostPattern = "?.XYZ.cn-northwest-1.rds.amazonaws.com.cn";
+    assertEquals(oldChinaExpectedHostPattern, target.getRdsInstanceHostPattern(oldChinaRegionCluster));
+    assertEquals(oldChinaExpectedHostPattern, target.getRdsInstanceHostPattern(oldChinaRegionClusterReadOnly));
+    assertEquals(oldChinaExpectedHostPattern, target.getRdsInstanceHostPattern(oldChinaRegionInstance));
+    assertEquals(oldChinaExpectedHostPattern, target.getRdsInstanceHostPattern(oldChinaRegionProxy));
+    assertEquals(oldChinaExpectedHostPattern, target.getRdsInstanceHostPattern(oldChinaRegionCustomDomain));
+  }
+
+  @Test
+  public void testIsRdsClusterDns() {
+    assertTrue(target.isRdsClusterDns(usEastRegionCluster));
+    assertTrue(target.isRdsClusterDns(usEastRegionClusterReadOnly));
+    assertFalse(target.isRdsClusterDns(usEastRegionInstance));
+    assertFalse(target.isRdsClusterDns(usEastRegionProxy));
+    assertFalse(target.isRdsClusterDns(usEastRegionCustomDomain));
+
+    assertTrue(target.isRdsClusterDns(chinaRegionCluster));
+    assertTrue(target.isRdsClusterDns(chinaRegionClusterReadOnly));
+    assertFalse(target.isRdsClusterDns(chinaRegionInstance));
+    assertFalse(target.isRdsClusterDns(chinaRegionProxy));
+    assertFalse(target.isRdsClusterDns(chinaRegionCustomDomain));
+
+    assertTrue(target.isRdsClusterDns(oldChinaRegionCluster));
+    assertTrue(target.isRdsClusterDns(oldChinaRegionClusterReadOnly));
+    assertFalse(target.isRdsClusterDns(oldChinaRegionInstance));
+    assertFalse(target.isRdsClusterDns(oldChinaRegionProxy));
+    assertFalse(target.isRdsClusterDns(oldChinaRegionCustomDomain));
+  }
+
+  @Test
+  public void testIsReaderClusterDns() {
+    assertFalse(target.isReaderClusterDns(usEastRegionCluster));
+    assertTrue(target.isReaderClusterDns(usEastRegionClusterReadOnly));
+    assertFalse(target.isReaderClusterDns(usEastRegionInstance));
+    assertFalse(target.isReaderClusterDns(usEastRegionProxy));
+    assertFalse(target.isReaderClusterDns(usEastRegionCustomDomain));
+
+    assertFalse(target.isReaderClusterDns(chinaRegionCluster));
+    assertTrue(target.isReaderClusterDns(chinaRegionClusterReadOnly));
+    assertFalse(target.isReaderClusterDns(chinaRegionInstance));
+    assertFalse(target.isReaderClusterDns(chinaRegionProxy));
+    assertFalse(target.isReaderClusterDns(chinaRegionCustomDomain));
+
+    assertFalse(target.isReaderClusterDns(oldChinaRegionCluster));
+    assertTrue(target.isReaderClusterDns(oldChinaRegionClusterReadOnly));
+    assertFalse(target.isReaderClusterDns(oldChinaRegionInstance));
+    assertFalse(target.isReaderClusterDns(oldChinaRegionProxy));
+    assertFalse(target.isReaderClusterDns(oldChinaRegionCustomDomain));
+  }
+
+  @Test
+  public void testBrokenPathsHostPattern() {
```

Comment:
NIT: i would suggest renaming broken for invalid or incorrect

Reply:
done